compiler/ir/backend.common/src/org/jetbrains/kotlin/ir/util/IrTypeUtils.kt

```diff
@@ -89,3 +92,38 @@ fun IrType.getPrimitiveArrayElementType() = (this as? IrSimpleType)?.let {
 
 fun IrType.isNonPrimitiveArray() =
     (this.isArray() || this.isNullableArray()) && !this.isPrimitiveArray()
+
+fun IrType.eraseTypeParameters() = when (this) {
```

Comment:
Please specify explicit return type, provide a few examples in the kdoc and probably rename function to emphasize what exactly this function does since it's not immediately obvious.

Also, the behavior seems very much JVM-specific, consider moving to JVM IR type utils.